src/DotNetProjectFile.Analyzers/Xml/Project.cs

```diff
@@ -0,0 +1,54 @@
+﻿using DotNetProjectFile.IO;
+using System.IO;
+using System.Xml.Linq;
+
+namespace DotNetProjectFile.Xml;
+
+public sealed class Project : Node
+{
+    public Project(XElement element) : base(element) { }
+
+    public Nodes<PropertyGroup> PropertyGroups => GetChildren<PropertyGroup>();
+
+    public Nodes<ItemGroup> ItemGroups => GetChildren<ItemGroup>();
+
+    public static Project Load(FileInfo file)
+        => new (XElement.Load(file.OpenRead(), LoadOptions.PreserveWhitespace | LoadOptions.SetLineInfo));
+    
+    public static Project? Create(Compilation compilation)
+    {
+        if (compilation.SourceModule.ContainingAssembly is { } assembly)
+        {
+            var directories = assembly.Locations.Select(l => l.SourceTree?.FilePath)
+                .Where(p => p is { })
+                .Select(p => new FileInfo(p).Directory)
+                .Distinct(DirectoryEqualityComparer.Instance)
+                .Where(f => f.Exists)
+                .ToArray();
+
+            var project = directories.Select(d => GetFile(d, $"{assembly.Name}.??proj"))
```

Comment:
What does this ??proj do?

Reply:
Matching both .csproj (C#) as vbproj (VB.NET). It is a bit lazy, so I could be improved on.